src/gd_kotlin.cpp

```diff
@@ -325,7 +340,65 @@ String GDKotlin::get_path_to_native_image() {
 
 void GDKotlin::unload_dynamic_lib() {
     if (OS::get_singleton()->close_dynamic_library(jvm_dynamic_library_handle) != OK) {
-        JVM_CRASH_NOW_MSG("Failed to close the jvm dynamic library!");
+        JVM_ERR_FAIL_MSG("Failed to close the jvm dynamic library!");
+    }
+}
+#endif
+
+#ifdef DEBUG_ENABLED
+void GDKotlin::validate_state() {
```

Comment:
I would probably move this to a separate class no? we could centralize the checks once we add more. Also for once we add back the editor plugin we could do some checks there.

But I'm also fine if we think about this later